### PR TITLE
Extend and correct documentation for lib/data_integrity_utils.pm

### PR DESCRIPTION
**Pull request by Zaoliang**
see https://progress.opensuse.org/issues/57449

### CODE REVIEW

File: lib/data_integrity_utils.pm [hunk 1]
```diff
@@ -22,8 +27,12 @@ use Digest::file 'digest_file_hex';
...
-=head2
+=head2 get_image_digest
+
+ get_image_digest();
+
 Returns image digest. Digest retrieval is platform-specific depends.
+
 =cut
 sub get_image_digest {
     my ($image_path) = shift;
```
Comment (SergioAtSUSE) [inline lib/data_integrity_utils.pm]: There is a parameter here, this needs to be mentioned in the documentation.

File: lib/data_integrity_utils.pm [hunk 2]
```diff
@@ -42,8 +51,12 @@ sub get_image_digest {
...
-Verify image data integrity by comparing SHA256 checksums
+
+ verify_checksum();
+
+Verify image data integrity by comparing SHA256 checksums.
 Returns error message in case of failure, empty string in case of success
+
 =cut
 sub verify_checksum {
     my ($dir_path) = shift;
```
Comment (SergioAtSUSE) [inline lib/data_integrity_utils.pm]: Same here: There is a parameter here, this needs to be mentioned in the documentation.

File: lib/data_integrity_utils.pm [hunk 3]
```diff
@@ -22,8 +27,13 @@ use Digest::file 'digest_file_hex';
 
 our @EXPORT = qw(verify_checksum get_image_digest);
 
-=head2
-Returns image digest. Digest retrieval is platform-specific depends.
+=head2 get_image_digest
+
+ get_image_digest();
```
Comment (SergioAtSUSE) [inline lib/data_integrity_utils.pm]:
```suggestion
 get_image_digest($image_path);
```

File: lib/data_integrity_utils.pm [hunk 4]
```diff
@@ -42,8 +52,14 @@ sub get_image_digest {
 }
 
 =head2 verify_checksum
-Verify image data integrity by comparing SHA256 checksums
-Returns error message in case of failure, empty string in case of success
+
+ verify_checksum();
```
Comment (SergioAtSUSE) [inline lib/data_integrity_utils.pm]:
```suggestion
 verify_checksum($dir_path);
```

File: lib/data_integrity_utils.pm [hunk 5]
```diff
@@ -22,8 +27,13 @@ use Digest::file 'digest_file_hex';
 
 our @EXPORT = qw(verify_checksum get_image_digest);
 
-=head2
-Returns image digest. Digest retrieval is platform-specific depends.
+=head2 get_image_digest
+
+ get_image_digest(image_path);
```
Comment (SergioAtSUSE) [inline lib/data_integrity_utils.pm]: Missing sign

```suggestion
 get_image_digest($image_path);
```

Comment (Zaoliang) [reply lib/data_integrity_utils.pm]: yes, updated now, thanks

File: lib/data_integrity_utils.pm [hunk 6]
```diff
@@ -22,8 +27,13 @@ use Digest::file 'digest_file_hex';
 
 our @EXPORT = qw(verify_checksum get_image_digest);
 
-=head2
-Returns image digest. Digest retrieval is platform-specific depends.
+=head2 get_image_digest
+
+ get_image_digest($image_path);
+
+Returns image digest. Image path '$image_path' is the parameter which is used to get image digest.
```
Comment (SergioAtSUSE) [inline lib/data_integrity_utils.pm]:
```suggestion
Returns image digest. Image path C<$image_path> is the parameter which is used to get image digest.
```

File: lib/data_integrity_utils.pm [hunk 7]
```diff
@@ -42,8 +52,14 @@ sub get_image_digest {
 }
 
 =head2 verify_checksum
-Verify image data integrity by comparing SHA256 checksums
-Returns error message in case of failure, empty string in case of success
+
+ verify_checksum($dir_path);
+
+Verify image data integrity by comparing SHA256 checksums.
+Directory path '$dir_path' is the parameter which is a part of image path '$image_path' if exists.
```
Comment (SergioAtSUSE) [inline lib/data_integrity_utils.pm]:
```suggestion
Directory path C<$dir_path> is the parameter which is a part of image path '$image_path' if exists.
```

**Pull request merged**